File: 공예진/programmers/P142086.java
//문제 링크: https://school.programmers.co.kr/learn/courses/30/lessons/142086
//시간: 46.12ms
//메모리: 86.8MB

import java.util.*;

class Solution {
    public int[] solution(String s) {
        int[] answer = new int[s.length()];
        Map<Character, Integer> map = new HashMap<>();

        for (int i = 0; i < s.length(); i++) {
            char ch = s.charAt(i);
            int num = map.getOrDefault(ch, -1);
            if (num == -1) {
                answer[i] = -1;
            } else{
                answer[i] = i - map.get(ch);
            }
            map.put(ch, i);
        }
        return answer;
    }
}

File: 공소연/programmers/P142086.js
// 링크: https://school.programmers.co.kr/learn/courses/30/lessons/142086?language=javascript
// 시간: 4.00ms
// 메모리: 37.9MB

function solution(s) {
    const map = {}
    return Array.from(s).map((c, idx) => {
        if (map[c]) {
            map[c] = [idx, idx - map[c][0]];
        } else {
            map[c] = [idx, -1]
        }
        return map[c][1];
    });
}
